src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -293,24 +310,39 @@ object TaskGroupBuilder {
   }
 
   private[this] def computeContainerInfo(
-    podVolumes: Seq[raml.Volume],
+    podVolumes: Seq[Volume],
     container: MesosContainer): Option[mesos.ContainerInfo.Builder] = {
 
     val containerInfo = mesos.ContainerInfo.newBuilder.setType(mesos.ContainerInfo.Type.MESOS)
 
     container.volumeMounts.foreach { volumeMount =>
-      podVolumes.find(_.name == volumeMount.name).map { hostVolume =>
-        val volume = mesos.Volume.newBuilder
-          .setContainerPath(volumeMount.mountPath)
-
-        hostVolume.host.foreach(volume.setHostPath)
-
-        // Read-write mode will be used when the "readOnly" option isn't set.
-        val mode = if (volumeMount.readOnly.getOrElse(false)) mesos.Volume.Mode.RO else mesos.Volume.Mode.RW
-
-        volume.setMode(mode)
 
-        containerInfo.addVolumes(volume)
+      // Read-write mode will be used when the "readOnly" option isn't set.
+      val mode = if (volumeMount.readOnly.getOrElse(false)) mesos.Volume.Mode.RO else mesos.Volume.Mode.RW
+
+      podVolumes.find(_.name == volumeMount.name).foreach {
```

Comment:
We have places where we reference other objects like volumes, endpoints etc.
Validation ensures, that this reference is always valid.
Perhaps it makes sense to add something like:
```
PodDefinition:
def volumeFor(mount: VolumeMount): Volume
```
which throws an exception in case of an invalid reference.
The way we handle this in a lot of places of the code is:
- complicated  (find and foreach)
- wrong (in case there is no volume this code is not executed